### PR TITLE
Fix Python 3.7 tests

### PR DESCRIPTION
**Pull request by bouweandela**
## Description

The syntax
```python
a = 1
f"{a=}"
```
is not supported in Python 3.7.

This pull request removes the use of that syntax.

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [ ] ~[☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do~

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#code-quality)
- [x] [🛠][1] [Unit tests](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#tests) have been added
- [x] [🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

**Comment by valeriupredoi** [comment]
cheers @bouweandela - just running a GA test, not that I don't trust you, but just to be extra sure :grin:

**Review by valeriupredoi — APPROVED** [review]
flying colors :airplane:

File: .github/workflows/action-test.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ on:
   push:
     branches:
     - main
-    - unpin_iris
+    - fix-python-3.7-tests
```
Comment (zklaus) [inline .github/workflows/action-test.yml:24]: Remove this line?

**Review by zklaus — CHANGES_REQUESTED** [review]
Looking good, thanks! Just don't forget to remove the branch from the tests before merging.

File: .github/workflows/action-test.yml (hunk 1 above)
Comment (valeriupredoi) [reply .github/workflows/action-test.yml]: yup, good catch I just cleaned up all the GA workflow files, some had older merged+deleted branches that were supposed to be running on :+1:

**Comment by valeriupredoi** [comment]
cheers for fixing @bouweandela :beer:

**Review by zklaus — APPROVED** [review]

**Pull request merged**